lib/oauther.ex

```diff
@@ -1,28 +1,46 @@
 defmodule OAuther do
   defmodule Credentials do
+    @type t :: %__MODULE__{
+      consumer_key:    String.t,
+      consumer_secret: String.t,
+      token:           String.t,
+      token_secret:    String.t,
+      method:          :hmac_sha1 | :rsa_sha1 | :plaintext
+    }
+
     defstruct [:consumer_key, :consumer_secret,
                :token, :token_secret, method: :hmac_sha1]
   end
 
+  @type pair_of_strings :: {String.t, String.t}
+  @type header :: {String.t, any}
+  @type headers :: [header]
```

Comment:
So this one is a bit tricky. Sure we only use the `header` type once but it's used in the `headers` and this one appears multiple times.
So if we get rid of the underlying `header` type, we'll end up with very repetitive and long specs.
Furthermore, I believe it adds a bit of semantic to the structure of the values returned which certainly help to understand the code.
However, if you insist, I will satisfy your request.